From e8a275348c66da69ac38fb97f87283c70cf63879 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Tue, 6 May 2025 14:13:42 +0300
Subject: [PATCH] arm64: dts: qcom: qcs9075-iq-9075-evk: enable UFS

Enable the UFS storage module on the Qualcomm iq-9075-evk board.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Pending [Waiting for iq-9075 patches to be resubmitted]
---
 .../boot/dts/qcom/qcs9075-iq-9075-evk.dts     | 30 +++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts b/arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts
index eadc59739a4b..c62fe976a91c 100644
--- a/arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts
+++ b/arch/arm64/boot/dts/qcom/qcs9075-iq-9075-evk.dts
@@ -15,6 +15,7 @@ / {
 
 	aliases {
 		serial0 = &uart10;
+		ufshc1 = &ufs_mem_hc;
 	};
 
 	chosen {
@@ -143,6 +144,18 @@ vreg_l3c: ldo3 {
 						   RPMH_REGULATOR_MODE_HPM>;
 		};
 
+		vreg_l4c: ldo4 {
+			regulator-name = "vreg_l4c";
+			regulator-min-microvolt = <1200000>;
+			regulator-max-microvolt = <1200000>;
+			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
+			/*
+			 * FIXME: This should have regulator-allow-set-load but
+			 * we're getting an over-current fault from the PMIC
+			 * when switching to LPM.
+			 */
+		};
+
 		vreg_l5c: ldo5 {
 			regulator-name = "vreg_l5c";
 			regulator-min-microvolt = <1100000>;
@@ -263,6 +276,23 @@ &uart10 {
 	status = "okay";
 };
 
+&ufs_mem_hc {
+	reset-gpios = <&tlmm 149 GPIO_ACTIVE_LOW>;
+	vcc-supply = <&vreg_l8a>;
+	vcc-max-microamp = <1100000>;
+	vccq-supply = <&vreg_l4c>;
+	vccq-max-microamp = <1200000>;
+
+	status = "okay";
+};
+
+&ufs_mem_phy {
+	vdda-phy-supply = <&vreg_l4a>;
+	vdda-pll-supply = <&vreg_l1c>;
+
+	status = "okay";
+};
+
 &xo_board_clk {
 	clock-frequency = <38400000>;
 };
